pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -555,6 +557,40 @@ func getContainer(el *v1alpha1.EventListener, c Config) corev1.Container {
 	}
 }
 
+func getPort(el *v1alpha1.EventListener, c Config) int {
```

Comment:
This whole function is not super nice / concise but it seems to me like the only way to safely determine which port to use.

Reply:
nice. the bit below about checking for cert Env seems repeated a couple of times...could we extract it to a function `isTLSEnabled` or something?

Reply:
I thought about this as well. Unfortunately we need to extract different information each time where this is done. I can try to refactor this later on but would prefer to do so in a separate PR.